app/models/settings.py

```diff
@@ -21,400 +21,61 @@ def __init__(self) -> None:
-    def todds_dry_run(self, value: bool) -> None:
-        if value == self._todds_dry_run:
-            return
-        self._todds_dry_run = value
-        EventBus().settings_have_changed.emit()
-
-    @property
-    def todds_overwrite(self) -> bool:
-        return self._todds_overwrite
-
-    @todds_overwrite.setter
-    def todds_overwrite(self, value: bool) -> None:
-        if value == self._todds_overwrite:
-            return
-        self._todds_overwrite = value
-        EventBus().settings_have_changed.emit()
-
-    @property
-    def current_instance(self) -> str:
-        return self._current_instance
-
-    @current_instance.setter
-    def current_instance(self, value: str) -> None:
-        if value == self._current_instance:
-            return
-        self._current_instance = value
-        EventBus().settings_have_changed.emit()
-
-    @property
-    def instances(self) -> dict[str, Instance]:
-        return self._instances
-
-    @instances.setter
-    def instances(self, value: dict[str, Instance]) -> None:
-        if value == self._instances:
-            return
-        self._instances = value
-        EventBus().settings_have_changed.emit()
-
-    @property
-    def github_username(self) -> str:
-        return self._github_username
-
-    @github_username.setter
-    def github_username(self, value: str) -> None:
-        if value == self._github_username:
-            return
-        self._github_username = value
-        EventBus().settings_have_changed.emit()
-
-    @property
-    def github_token(self) -> str:
-        return self._github_token
-
-    @github_token.setter
-    def github_token(self, value: str) -> None:
-        if value == self._github_token:
-            return
-        self._github_token = value
-        EventBus().settings_have_changed.emit()
-
-    @property
-    def steam_apikey(self) -> str:
-        return self._steam_apikey
-
-    @steam_apikey.setter
-    def steam_apikey(self, value: str) -> None:
-        if value == self._steam_apikey:
-            return
-        self._steam_apikey = value
+        self.external_community_rules_metadata_source: str = "None"
+        self.db_builder_include: str = "all_mods"
+        self.database_expiry: int = 604800
+        self.build_steam_database_dlc_data: bool = True
+        self.build_steam_database_update_toggle: bool = False
+        self.watchdog_toggle: bool = True
+        self.mod_type_filter_toggle: bool = True
+        self.duplicate_mods_warning: bool = False
+        self.steam_mods_update_check: bool = False
+        self.try_download_missing_mods: bool = False
+        self.steamcmd_validate_downloads: bool = True
+        self.todds_preset: str = "optimized"
+        self.todds_active_mods_target: bool = True
+        self.todds_dry_run: bool = False
+        self.todds_overwrite: bool = False
+        self.current_instance: str = "Default"
+        self.instances: dict[str, Instance] = {"Default": Instance()}
```

Comment:
just move 
self.instances: dict[str, Instance] = {"Default": Instance()}
below
self.steam_apikey: str = ""
since all instances are added below that in the json

Reply:
I don't think the order should matter? The only thing it affects is the order of writing when dumping to json. Reading has no effect since the json file is just converted to a dictionary anyways.

Reply:
agreed but it looks nice in settings.json